Source/WebCore/PAL/pal/text/TextCodecCJK.cpp

```diff
@@ -855,19 +855,19 @@ static const std::array<std::pair<uint32_t, char32_t>, 207>& gb18030Ranges()
     return ranges;
 }
 
-WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
-
 // https://encoding.spec.whatwg.org/#index-gb18030-ranges-code-point
 static std::optional<char32_t> gb18030RangesCodePoint(uint32_t pointer)
 {
     if ((pointer > 39419 && pointer < 189000) || pointer > 1237575)
         return std::nullopt;
     if (pointer == 7457)
         return 0xE7C7;
-    auto upperBound = std::ranges::upper_bound(gb18030Ranges(), makeFirstAdapter(pointer), CompareFirst { });
-    ASSERT(upperBound != gb18030Ranges().begin());
-    uint32_t offset = (upperBound - 1)->first;
-    char32_t codePointOffset = (upperBound - 1)->second;
+    auto& ranges = gb18030Ranges();
+    auto upperBound = std::ranges::upper_bound(ranges, makeFirstAdapter(pointer), CompareFirst { });
+    ASSERT(upperBound != ranges.begin());
+    auto& range = ranges[upperBound - ranges.begin() - 1];
+    uint32_t offset = range.first;
+    char32_t codePointOffset = range.second;
```

Comment:
Does something like this work?
```suggestion
    auto [offset, codePointOffset] = ranges[upperBound - ranges.begin() - 1];
```

Reply:
Yes it does. I'll update the PR.